src/modules/core/components/DetailsWidgetUser/DetailsWidgetUser.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import HookedUserAvatar from '~users/HookedUserAvatar';
+import MaskedAddress from '~core/MaskedAddress';
+import InvisibleCopyableAddress from '~core/InvisibleCopyableAddress';
+import { useUser } from '~data/index';
+import { Address } from '~types/index';
+
+import styles from './DetailsWidgetUser.css';
+
+const displayName = 'DetailsWidgetUser';
+
+interface Props {
+  walletAddress: Address;
+}
+
+const DetailsWidgetUser = ({ walletAddress }: Props) => {
+  const UserAvatar = HookedUserAvatar({ fetchUser: false });
+  const userProfile = useUser(walletAddress);
+  const userDisplayName = userProfile?.profile?.displayName;
+  const username = userProfile?.profile?.username;
+
+  return (
+    <div className={styles.main}>
+      <UserAvatar
+        size="s"
+        notSet={false}
+        address={walletAddress || ''}
+        showInfo
+      />
+      {(userDisplayName || username) && (
```

Comment:
Theoretically this logic check is wrong, since this will pass if the `userDisplayName` is available.

But since you won't have a `userDisplayName` without a `username`, it's all good